sdb/commands/help.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2019 Chuck Tuffli
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class Help(sdb.Command):
+    # pylint: disable=too-few-public-methods
+
+    "Print command usage"
+
+    names = ["help"]
+
+    def _init_argparse(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument('-v', '--verbose', action='store_true')
+        parser.add_argument('cmd', nargs='?')
+
+    def call(self, objs: Iterable[drgn.Object]):
+        if self.args.cmd:
+            if self.args.cmd in sdb.all_commands:
+                cmd_list = [self.args.cmd]
+            else:
+                print('Unknown command: ' + self.args.cmd)
+                return
+        else:
```

Comment:
I'm on the fence w.r.t. us supporting "help" without any arguments.. For example, I can perhaps see utility in using it to get a list of all possible commands:
```
> help ! head -n 5
filter: Returns a list of objects matching the given expression
stacks: This is the superclass of all SDB command classes.
help: Print command usage
cast: This is the superclass of all SDB command classes.
walk: This is the superclass of all SDB command classes.
```

But at the same time, if I use `-v` it becomes difficult to read (mainly due to the lack of formatting):
```
> help -v ! head
filter: Returns a list of objects matching the given expression
stacks: This is the superclass of all SDB command classes.

This class intends to be the superclass of all other SDB command
classes, and is responsible for implementing all the logic that is
required to integrate the command with the SDB REPL.
help: Print command usage
cast: This is the superclass of all SDB command classes.

This class intends to be the superclass of all other SDB command
> help -v ! head -n 20
filter: Returns a list of objects matching the given expression
stacks: This is the superclass of all SDB command classes.

This class intends to be the superclass of all other SDB command
classes, and is responsible for implementing all the logic that is
required to integrate the command with the SDB REPL.
help: Print command usage
cast: This is the superclass of all SDB command classes.

This class intends to be the superclass of all other SDB command
classes, and is responsible for implementing all the logic that is
required to integrate the command with the SDB REPL.
walk: This is the superclass of all SDB command classes.

This class intends to be the superclass of all other SDB command
classes, and is responsible for implementing all the logic that is
required to integrate the command with the SDB REPL.
pretty_print: This is the superclass of all SDB command classes.
aliases: ['pp']

```

Further, we can also get a list of all commands by hitting "Tab" twice, and leveraging readline's functionality:
```
>
addr              count             member            spa               walk
address           dmesg             metaslab          spl_kmem_caches   wc
arc               echo              pp                spl_list          zfs_dbgmsg
avl               exit              pretty_print      stacks
cast              filter            pyfilter          tail
cc                head              quit              type
cnt               help              slabs             vdev
```

I'm curious how others feel about this. Should we require the user to pass in a command name to `help`?

Reply:
The behavior of help without arguments was in response to a question at the hackathon as to what commands existed. Typing in `help` might be more intuitive then hitting tab twice for some users.
As for the behavior of `help -v`, I agree that the output is unwieldy and perhaps not worth it.

Reply:
The double-tab was also nonintuitive to me, I had to ask @sdimitro.  I agree that it would be useful for `help` with no args to print what commands are available.